flink-examples/flink-examples-table/src/main/java/org/apache/flink/table/examples/java/connectors/SocketDynamicTableSource.java

```diff
@@ -69,16 +73,28 @@ public ChangelogMode getChangelogMode() {
 
     @Override
     public ScanRuntimeProvider getScanRuntimeProvider(ScanContext runtimeProviderContext) {
+        return new DataStreamScanProvider() {
+            @Override
+            public DataStream<RowData> produceDataStream(
+                    ProviderContext providerContext, StreamExecutionEnvironment execEnv) {
+                final DeserializationSchema<RowData> deserializer =
+                        decodingFormat.createRuntimeDecoder(
+                                runtimeProviderContext, producedDataType);
 
-        // create runtime classes that are shipped to the cluster
+                final SocketSource socketSource =
+                        new SocketSource(hostname, port, byteDelimiter, deserializer);
 
-        final DeserializationSchema<RowData> deserializer =
-                decodingFormat.createRuntimeDecoder(runtimeProviderContext, producedDataType);
+                return execEnv.fromSource(
+                                socketSource, WatermarkStrategy.noWatermarks(), "SocketSource")
+                        // SocketSource can only work with a parallelism of 1.
```

Comment:
we'd better add the description of only supporting parallelism 1 to the Class Java doc explicitly.

Reply:
Good sugesstion, added it.